plugins/inputs/linux_cpu/linux_cpu.go

```diff
@@ -0,0 +1,226 @@
+	var cpus []cpu
+
+	glob := path.Join(g.PathSysfs, "devices/system/cpu/cpu[0-9]*")
+	cpuDirs, err := filepath.Glob(glob)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(cpuDirs) == 0 {
+		return nil, fmt.Errorf("no CPUs detected at: %s", glob)
+	}
+
+	for _, dir := range cpuDirs {
+		_, cpuName := filepath.Split(dir)
+		cpuNum := strings.TrimPrefix(cpuName, "cpu")
+
+		cpu := cpu{
+			id:    cpuNum,
+			path:  dir,
+			props: make(map[string]string),
+		}
+
+		var props []prop
+
+		if choice.Contains(cpufreq, g.Metrics) {
+			props = append(props,
+				prop{name: "scaling_cur_freq", path: "cpufreq/scaling_cur_freq", optional: false},
+				prop{name: "scaling_min_freq", path: "cpufreq/scaling_min_freq", optional: false},
+				prop{name: "scaling_max_freq", path: "cpufreq/scaling_max_freq", optional: false},
+				prop{name: "cpuinfo_cur_freq", path: "cpufreq/cpuinfo_cur_freq", optional: true},
+				prop{name: "cpuinfo_min_freq", path: "cpufreq/cpuinfo_min_freq", optional: true},
+				prop{name: "cpuinfo_max_freq", path: "cpufreq/cpuinfo_max_freq", optional: true},
+			)
+		}
+
+		if choice.Contains(thermal, g.Metrics) {
+			props = append(
+				props,
+				prop{name: "throttle_count", path: "thermal_throttle/core_throttle_count", optional: false},
+				prop{name: "throttle_max_time", path: "thermal_throttle/core_throttle_max_time_ms", optional: false},
+				prop{name: "throttle_total_time", path: "thermal_throttle/core_throttle_total_time_ms", optional: false},
+			)
+		}
+
+		var failed = false
+		for _, prop := range props {
+			propPath := filepath.Join(dir, prop.path)
+			err := validatePath(propPath)
+			if err != nil {
+				if prop.optional {
+					continue
+				}
+
+				g.Log.Warnf("Failed to load property %s: %s", filepath.Join(dir, prop.path), err)
+				failed = true
+				break
+			}
+
+			cpu.props[prop.name] = propPath
+		}
+
+		if len(cpu.props) == 0 {
+			g.Log.Warnf("No properties enabled for CPU %s", cpuNum)
+			failed = true
+		}
+
+		if !failed {
+			cpus = append(cpus, cpu)
+		}
+	}
+	return cpus, nil
+}
+
+func init() {
+	inputs.Add("linux_cpu", func() telegraf.Input {
+		return &LinuxCPU{
+			Metrics: []string{"cpufreq", "thermal"},
+		}
+	})
+}
+
+func validatePath(propPath string) error {
+	f, err := os.Open(propPath)
+
+	if os.IsNotExist(err) {
+		return fmt.Errorf("CPU property does not exist: [%s]", propPath)
+	}
+
+	if err != nil {
+		return fmt.Errorf("cannot get system information for CPU property: [%s] - %v", propPath, err)
+	}
+
+	_ = f.Close() // File is not written to, closing should be safe
+	return nil
+}
+
+func readUintFromFile(propPath string) (uint64, error) {
+	f, err := os.Open(propPath)
+	if err != nil {
+		return 0, err
+	}
+	defer f.Close()
+
+	buffer := make([]byte, 22)
+
+	n, err := f.Read(buffer)
+	if err != nil && err != io.EOF {
+		return 0, fmt.Errorf("error on reading file, err: %v", err)
+	} else if n == 0 {
+		return 0, nil
+	}
+
+	return strconv.ParseUint(string(buffer[:n-1]), 10, 64)
```

Comment:
Maybe this would be a little bit cleaner?
```
	read, err := io.ReadAll(f)
	if err != nil {
		return 0, err
	}
	
	trimmedString := strings.TrimRight(string(read), "\n")
	return strconv.ParseUint(trimmedString, 10, 64)
```

Reply:
My preference would be the first approach, since we know that the input cannot exceed 22 bytes if the value is a `uint`, so we don't need to allocate a dynamic buffer.

Reply:
Ok, will not insist. Do you want to return `0` as a value when file was read by was empty?

Reply:
Hmm, good question, perhaps that should also give an error. What do you think?

Reply:
I think that missing value probably will not mean that 0 should be used as a default value instead. So error could be better option.